packages/zoe/test/unitTests/contracts/loan/test-liquidate.js

```diff
@@ -0,0 +1,116 @@
+// ts-check
+
+import '../../../../exported';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import '@agoric/install-ses';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import test from 'ava';
+
+import { doLiquidation } from '../../../../src/contracts/loan/liquidate';
+
+import {
+  setupLoanUnitTest,
+  makeSeatKit,
+  checkNoNewOffers,
+  checkPayouts,
+} from './helpers';
+
+import { trade } from '../../../../src/contractSupport';
+
+test('test doLiquidation with mocked autoswap', async t => {
```

Comment:
additional test:

 * autoswap fails
 * multiple scheduled liquidations don't interfere

Reply:
'Multiple scheduled liquidations' is already tested. 

For 'autoswap fails' - do you mean the trade fails? That is not tested.

Reply:
Sorry, I don't see a test that verifies that the two liquidations from an initial borrow and adding collateral don't interfere. Which test are you referring to?

> For 'autoswap fails' - do you mean the trade fails? 

Yeah, with autoswap, that pair might not have been initialized, or all the collateral could have been removed. There's no guarantee that autoswap will always succeed. Particularly in the start-up sequence, it's conceivable that a pair of currencies could be missing.

Reply:
> Sorry, I don't see a test that verifies that the two liquidations from an initial borrow and adding collateral don't interfere. Which test are you referring to?

 'borrow, then addCollateral, then getLiquidationPromise'